### ISSUE TITLE
Add git command console/terminal

### ISSUE THREAD

**Issue by kortschak** [sublimehq/sublime_merge#201]
**Problem description**

There are a number of git commands that do not yet exist, and may not ever exist given the wide world of git. Many of these commands are very useful (git rebase -i, git bisect...).

**Preferred solution**

Add git console similar to sublime texts <kbd>CTRL-`</kbd> console.

**Alternatives**

Commands to open system terminal at the current file and/or repo base.

**Comment by deathaxe** [comment]
related to #186

**Comment by TeamSpen210** [comment]
An alternative I was thinking of suggesting would be to simply add a `git` command to the command palette which prompts/takes an argument, then executes that directly and displays output in the current output dropdown. That would probably be more convienient for non-interactive commands, and allows you to bind them in menus/keymaps/etc.

**Comment by deathaxe** [comment]
@TeamSpen210: This command actually exists already.

```
{
  "caption": "Develop, switch to the develop branch and run a update/fetch",
  "command": "git",
  "args": { "argv": ["dev"] }
}
```

**Comment by kortschak** [comment]
@deathaxe, no this is not the same.Your command there has a fixed argument, it does not prompt for one.

**Comment by jasonycw** [comment]
I think @TeamSpen210 is talking about a place to do git command directly, maybe an integrated git command panel like Sublime Text
![image](https://user-images.githubusercontent.com/4518597/47842736-8ce13f00-ddf8-11e8-97fe-b9a0f7bd05fc.png)

SourceTree has a button to open Git bash for that repository directly
1. Just click here<br>![image](https://user-images.githubusercontent.com/4518597/47842783-a7b3b380-ddf8-11e8-84ab-66c119dc78ec.png)
2. And this pop up, the directory is already point to the repo folder<br>![image](https://user-images.githubusercontent.com/4518597/47842819-bac68380-ddf8-11e8-9000-da04ae3264d2.png)

**Comment by kortschak** [comment]
Yes. This also has the advantage of putting you in the repo that you want to manipulate. You could, I imagine get a terminal to open in the correct place, but that does not yet work, e.g.:
```
   {
        "caption": "Open terminal",
        "command": "/usr/bin/gnome-terminal",
        "args": { "--working-directory": ["$working_dir"] }
    },
```

Unless there is an internal command that will run an arbitrary executable that I've not found yet. The only shelling out command that I can see is "open_in_editor" which just runs the executable that's specified in the editor path.

**Comment by CheyenneWills** [comment]
Like to see something like this as well.  In my case For example, I have a local git subcommand that takes a parameter (for example a ticket # in a local tracking system).  In a terminal command line, the command would look something like:
  ` git fetch-ticket 1234
`
I've tried building a command:
```
[
   {
      "caption": "Fetch Ticket",
      "command": "git",
      "args": { "argv": ["fetch-ticket",
                                 ???? << need something here that I can fill in via a prompt
                               ]
              }
   }
]

```

**Comment by wbond** [comment]
A git command palette entry has been discussed, but runs into issues where many git commands require input, so would explicit support for every command and argument, it would likely that git would get "stuck" waiting for input.

The idea of a way to open a terminal is discussed some at https://github.com/sublimehq/sublime_merge/issues/186#issuecomment-450894407.

**Comment by ratijas** [comment]
Talking about Linux here,

In KDE world, KParts module provide a great deal of functionality including Konsole plugin — which is how it works in Dolphin, for example.

![Dolphin](https://user-images.githubusercontent.com/6737986/123549711-cf5aa580-d772-11eb-8377-637a8d6c9cde.png)

Besides that, KDE has necessary infrastructure support for choosing default terminal app via GUI settings.

![KDE system settings: default terminal](https://user-images.githubusercontent.com/6737986/123549767-134daa80-d773-11eb-9fdb-52425d2000b4.png)

But in GTK world… uhh… it's messy.  Up to the point where they have a giant [if-else-if train](https://gitlab.gnome.org/GNOME/glib/-/blob/c1a11c02e5ef458f328c1ed938d6077a4fd7deb3/gio/gdesktopappinfo.c#L2548-2586) checking for every other known program on a PATH. Looking for a [conceptually right solution](https://gitlab.gnome.org/GNOME/glib/-/issues/338) to this problem takes them literally more that a decade already.

**Comment by dpjohnst** [comment]
G'day all,

We've just added _Open in Terminal_ to the command palette and action menu in Sublime Merge Build 2115 (Development Build).

I appreciate that this isn't quite what's requested here, but it's worth noting.

Cheers,
Dylan from Sublime HQ
